src/interface/swarm.cpp

```diff
@@ -317,28 +354,469 @@ void Swarm::Defrag() {
+      KOKKOS_LAMBDA(const int n) {        // Max index
+        for (int m = 0; m < nbmax; m++) { // Number of neighbors
+          if (n < num_particles_to_send(m)) {
+            int sidx = particle_indices_to_send(m, n);
+            int buffer_index = n * particle_size;
+            swarm_d.MarkParticleForRemoval(sidx);
+            for (int i = 0; i < real_vars_size; i++) {
+              bdvar.send[m](buffer_index) = vreal(i, sidx);
+              buffer_index++;
+            }
+            for (int i = 0; i < int_vars_size; i++) {
+              bdvar.send[m](buffer_index) = static_cast<Real>(vint(i, sidx));
+              buffer_index++;
+            }
+            // If rank is shared, apply boundary conditions here
+            // TODO(BRR) Don't hardcode periodic boundary conditions
+            if (nrank(m) == swarm_d.GetMyRank()) {
+              double &x = vreal(ix, sidx);
+              double &y = vreal(iy, sidx);
+              double &z = vreal(iz, sidx);
+              if (x < swarm_d.x_min_global_) {
+                x = swarm_d.x_max_global_ - (swarm_d.x_min_global_ - x);
+              }
+              if (x > swarm_d.x_max_global_) {
+                x = swarm_d.x_min_global_ + (x - swarm_d.x_max_global_);
+              }
+              if (y < swarm_d.y_min_global_) {
+                y = swarm_d.y_max_global_ - (swarm_d.y_min_global_ - y);
+              }
+              if (y > swarm_d.y_max_global_) {
+                y = swarm_d.y_min_global_ + (y - swarm_d.y_max_global_);
+              }
+              if (z < swarm_d.z_min_global_) {
+                z = swarm_d.z_max_global_ - (swarm_d.z_min_global_ - z);
+              }
+              if (z > swarm_d.z_max_global_) {
+                z = swarm_d.z_min_global_ + (z - swarm_d.z_max_global_);
+              }
+            }
+          }
+        }
+      });
+
+  // Count all the particles that are Active and Not on this block, if nonzero,
+  // copy into buffers (if no send already for that buffer) and send
+
+  RemoveMarkedParticles();
+
+  vbswarm->Send(phase);
+  return true;
+}
+
+vpack_types::SwarmVarList<Real> Swarm::MakeRealList_(std::vector<std::string> &names) {
+  int size = 0;
+  vpack_types::SwarmVarList<Real> vars;
+
+  for (auto it = realVector_.rbegin(); it != realVector_.rend(); ++it) {
+    auto v = *it;
+    vars.push_front(v);
+    size++;
+  }
+  // Get names in same order as list
+  names.resize(size);
+  int it = 0;
+  for (auto &v : vars) {
+    names[it++] = v->label();
+  }
+  return vars;
+}
+
+vpack_types::SwarmVarList<int> Swarm::MakeIntList_(std::vector<std::string> &names) {
+  int size = 0;
+  vpack_types::SwarmVarList<int> vars;
+
+  for (auto it = intVector_.rbegin(); it != intVector_.rend(); ++it) {
+    auto v = *it;
+    vars.push_front(v);
+    size++;
+  }
+  // Get names in same order as list
+  names.resize(size);
+  int it = 0;
+  for (auto &v : vars) {
+    names[it++] = v->label();
+  }
+  return vars;
+}
```

Comment:
The logic here is identical can you make it a template?

Reply:
It also does not appear you need the logic with names[it++] in the second for loop I think you can delete that. In the code below you never use expanded_names

Reply:
Unless I'm missing somewhere that you are calling it.

Reply:
You're right, right now we only ever use this to pack all `ParticleVariable`s and we don't look them up by their names so `expanded_names` is irrelevant. I borrowed this logic from elsewhere and left some of it in in case we someday want to pack specific subsets of variables, but I don't see a use case for this at the moment -- so I'll just remove the unused logic